dateparser/date_parser.py

```diff
@@ -32,10 +34,65 @@ def __add__(self, other):
 parser.relativedelta.relativedelta = new_relativedelta
 
 
+class new_parser(parser.parser):
+    """
+    Implements an alternate parse method which supports preference to dates in future and past.
+    For more see issue #36
+    """
+
+    def parse(self, timestr, default=None, ignoretz=False, prefer_future=None, **kwargs):
+        # timestr needs to be a buffer as required by _parse
+        timestr = timestr if not isinstance(timestr, str) else StringIO(timestr)
```

Comment:
https://github.com/dateutil/dateutil/blob/master/dateutil/parser.py#L908

I changed my StringIO line into that exact line.